File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        Map<Character, Character> map1 = new HashMap<>();
        if (s.length() != t.length()) {
            return false;
        }
        for (int i = 0; i < s.length(); i++) {
            char ch1 = s.charAt(i);
            char ch2 = t.charAt(i);
            if (map1.containsKey(ch1)) {
                if (map1.get(ch1) != ch2) {
                    return false;
                }
            } else {
                if (map1.containsValue(ch2)) {
                    return false;
                } else {
                    map1.put(ch1, ch2);
                }
            }
        }
        return true;
    }
}

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if len(s)!=len(t):
            return False
        map={}
        mapped_values=set()

        for i in range(len(s)):
            ch1=s[i]
            ch2=t[i]
            if ch1 in map:
                if map[ch1]!=ch2:
                    return False
            else:
                if ch2 in mapped_values:
                    return False
                else:
                    map[ch1]=ch2
                    mapped_values.add(ch2)

        return True